src/pathfinding_service/database.py

```diff
@@ -227,3 +227,14 @@ def get_feedback_token(
             )
 
         return None
+
+    def get_num_routes_feedback(
+        self, only_with_feedback: bool = False, only_successful: bool = False
+    ) -> int:
+        where_clause = ""
+        if only_with_feedback:
+            where_clause = "WHERE successful IS NOT NULL"
+        elif only_successful:
+            where_clause = "WHERE successful"
+
+        return self.conn.execute(f"SELECT COUNT(*) FROM feedback {where_clause};").fetchone()[0]
```

Comment:
If you wanted to do all these in a single pass through the table, you could do
```sql
SELECT
    count(*) AS all
    count(CASE WHEN feedback IS NOT NULL THEN 1 END) AS with_feedback,
    count(CASE WHEN feedback THEN 1 END) AS success
FROM feedback
```
Just because you asked before, not because any change is needed, now.